addons/spreadsheet_account/tests/test_debit_credit.py

```diff
@@ -101,6 +118,186 @@ def setUpClass(cls):
+                'company_id': None,
+            },
+            {
+                'date_range': _get_cell_date(2021),
+                'company_id': None,
+            },
+        ]
+        self.env['account.account']._pre_process_date_period_boundaries(cells)
+        timeline = self.env['account.account']._get_timeline(cells)
+        self.assertListEqual(timeline, [
+            (date(1900, 1, 1), date(2020, 12, 31)),
+            (date(2021, 1, 1), date(2021, 3, 31)),
+            (date(2021, 4, 1), date(2021, 5, 31)),
+            (date(2021, 6, 1), date(2021, 6, 30)),
+            (date(2021, 7, 1), date(2021, 12, 31)),
+        ])
+
+    def test_get_timeline_following_periods(self):
+        cells = [
+            {
+                'date_range': _get_cell_date(2022, quarter=2),
+                'company_id': None,
+            },
+            {
+                'date_range': _get_cell_date(2022, quarter=3),
+                'company_id': None,
+            },
+        ]
+        self.env['account.account']._pre_process_date_period_boundaries(cells)
+        timeline = self.env['account.account']._get_timeline(cells)
+        self.assertListEqual(timeline, [
+            (date(1900, 1, 1), date(2022, 3, 31)),
+            (date(2022, 4, 1), date(2022, 6, 30)),
+            (date(2022, 7, 1), date(2022, 9, 30)),
+        ])
+    
+    def test_get_timeline_following_periods_in_broader_period(self):
+        cells = [
+            {
+                'date_range': _get_cell_date(2022, quarter=2),
+                'company_id': None,
+            },
+            {
+                'date_range': _get_cell_date(2022, quarter=3),
+                'company_id': None,
+            },
+            {
+                'date_range': _get_cell_date(2022),
+                'company_id': None,
+            },
+        ]
+        self.env['account.account']._pre_process_date_period_boundaries(cells)
+        timeline = self.env['account.account']._get_timeline(cells)
+        self.assertListEqual(timeline, [
+            (date(1900, 1, 1), date(2021, 12, 31)),
+            (date(2022, 1, 1), date(2022, 3, 31)),
+            (date(2022, 4, 1), date(2022, 6, 30)),
+            (date(2022, 7, 1), date(2022, 9, 30)),
+            (date(2022, 10, 1), date(2022, 12, 31)),
+        ])
+    
+    def test_get_timeline_gapped_periods(self):
+        cells = [
+            {
+                'date_range': _get_cell_date(2022),
+                'company_id': None,
+            },
+            {
+                'date_range': _get_cell_date(2024),
+                'company_id': None,
+            }
+        ]
+        self.env['account.account']._pre_process_date_period_boundaries(cells)
+        timeline = self.env['account.account']._get_timeline(cells)
+        self.assertListEqual(timeline, [
+            (date(1900, 1, 1), date(2021, 12, 31)),
+            (date(2022, 1, 1), date(2022, 12, 31)),
+            (date(2024, 1, 1), date(2024, 12, 31)),
```

Comment:
Isn't there a unwanted gap if you have accounts with include initial balance?

Reply:
Indeed, nicely spotted, I forgot about it when changing it with your last comment ahah. I'll work on that :smile: